### ISSUE TITLE
 Trouble with REST API

### ISSUE THREAD

**Issue by ckuenne** [RotherOSS/otobo#2780]
Hi, I have a probably very basic question considering the OTOBO 10.1 REST/POST API, which I fail to get running. How do i get TicketSearch to work with a minimal example using curl? I just want to get any tickets of queue "Bioinformatics" for the beginning.

Our admins have already set up an OTOBO webservice called "TicketSearch" leading to url: https://XY/otobo/nph-genericinterface.pl/Webservice/TicketSearch/TicketSearch

I tried to follow the documentation (https://doc.otobo.org/manual/admin/10.1/en/content/processes-automation/web-services.html#operations) which points to the source code of TicketSearch itself (https://github.com/RotherOSS/otobo/blob/rel-10_1/Kernel/GenericInterface/Operation/Ticket/TicketSearch.pm#L70). But that is kinda hard to follow, since it's not obvious which parameters are actually mandatory.

Example:

file request.json:
```
{
        "Queues": "Bioinformatics"
}
```

cmd:
```
curl -v -k -X POST --header "Content-Type: application/json" -d @request.json https://XY/otobo/nph-genericinterface.pl/Webservice/TicketSearch/TicketSearch
```

cmd output:
```
Note: Unnecessary use of -X or --request, POST is already inferred.
*   Trying IP-XY:443...
* Connected to XY (IP-XY) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: ...
*  start date: Aug 30 07:16:00 2022 GMT
*  expire date: Aug 30 07:16:00 2032 GMT
*  issuer: ...
*  SSL certificate verify result: unable to get local issuer certificate (20), continuing anyway.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x555601b00b20)
> POST /otobo/nph-genericinterface.pl/Webservice/TicketSearch/TicketSearch HTTP/2
> Host: XY
> user-agent: curl/7.74.0
> accept: */*
> content-type: application/json
> content-length: 29
>
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
* Connection state changed (MAX_CONCURRENT_STREAMS == 128)!
* We are completely uploaded and fine
< HTTP/2 500
< server: nginx/1.25.2
< date: Mon, 18 Dec 2023 15:12:01 GMT
< content-length: 0
<
* Connection #0 to host XY left intact
```

I have also tried to include UserLogin/Password in the request or to put the json data as curl parameter --data, which did not work either.

How can I figure this out or at least exclude possible issues? I should mention that I will only have client-side access for the next couple of weeks, because holidays.

**Comment by svenoe** [comment]
Hi ckuenne,

please visit our [forums](https://otobo.de/de/forums/) with general questions and problems for community support. Github is for development issues, only. (For the web services you would need to set them up in the web interface (Admin->Webservices) before, in any case - not sure whether you did that.)

Thanks for your understanding, Sven
